pkg/pcr/measurement.go

```diff
@@ -248,6 +279,31 @@ func NewRangeMeasurement(id MeasurementID, offset uint64, length uint64) *Measur
 	}
 }
 
+// CachedMeasurement is a Measurement with hash value computed at creation time
+type CachedMeasurement struct {
+	Measurement
+	data []byte
+	hash []byte
+}
+
+func (m Measurement) Cache(image []byte, hasher hash.Hash) (*CachedMeasurement, error) {
```

Comment:
It would be more generic to have

NewCacheMeasurement(me MeasurementEvent) (*CachedMeasurement, error)

In that case it could cache anything that matches MeasurementEvent, not just Measurement